src/nunavut/lang/cpp/templates/deserialization.j2

```diff
@@ -183,11 +183,11 @@
         {% set ref_size = 'size'|to_template_unique_name %}
         // Array length prefix: {{ t.length_field_type }}
         {{ _deserialize_integer(t.length_field_type, ('const %s %s'|format((t.length_field_type | declaration), ref_size)) , offset) }}
-        if ( {{ ref_size}} > {{ t.capacity }}U)
+        if ( {{ ref_size }} > {{ t.capacity }}U)
```

Comment:
changes to the serialization logic to use reserve which is lighter-weight since no object initialization occurs within this call.